packages/aragon-cli/src/lib/configEnvironment.ts

```diff
@@ -0,0 +1,203 @@
+  apm: ApmConfig
+  // Objects injected from other middlewares
+  module?: ArappModule
+  reporter: any
+}
+
+interface EnvironmentsConfig {
+  [environmentName: string]: EnvironmentConfig
+}
+
+interface EnvironmentConfig {
+  registry: string // Registry contract address
+  appName?: string // ENS domain of the app
+  network: string // Network to use
+  wsRPC: string // Node WS URL
+  apm?: ApmConfig // "gateway": "https://ipfs.eth.aragon.network/ipfs"
+}
+
+interface AragonAppConfigurationFile {
+  path?: string // The path to the main contract in your app.
+  roles?: {
+    // An array of all the roles that your app has.
+    id: string // The identifier of the role as it is defined in the contract.
+    name: string // A description of the role in the app.
+    params: string[] // The names of any parameters for the role.
+  }[]
+  environments: EnvironmentsConfig // An object containing deploy environment configurations.
+}
+
+// Mutations
+interface ArappModule extends AragonAppConfigurationFile {
+  env?: EnvironmentConfig // Choosen configuration
+  appName?: string
+}
+
+function getEnv(arapp?: ArappModule, environment?: string): EnvironmentConfig {
+  if (arapp) {
+    if (!environment) environment = 'default'
+    const env = arapp.environments[environment]
+    if (!env)
+      throw Error(
+        `${environment} environment was not defined in your arapp.json.`
+      )
+    return env
+  }
+
+  // if there is no arapp.json and the command is not init default to the "global" config
+  // designed for the dao commands including dao acl
+  else {
+    if (!environment) environment = 'aragon:local'
+    const env = defaultEnvironments[environment]
+    if (!env)
+      throw Error(
+        `Could not find the ${environment} environment. Try using aragon:local, aragon:rinkeby or aragon:mainnet.`
+      )
+    return env
+  }
+}
+
+function configureNetwork(
+  network: string,
+  truffleConfig: TruffleConfig,
+  options?: { useFrame?: boolean; ignoreNetwork: boolean }
+) {
+  const { useFrame, ignoreNetwork } = options || {}
+
+  if (ignoreNetwork) return {}
+
+  if (useFrame) {
+    return {
+      name: `frame-${network}`,
+      provider: new Web3.providers.WebsocketProvider(
+        FRAME_ENDPOINT,
+        { headers: { origin: FRAME_ORIGIN } } // Provider options
+      ),
+    }
+  }
+
+  const truffleNetwork = truffleConfig.networks[network]
+  if (!truffleNetwork) {
+    throw new Error(
+      `aragon <command> requires a network '${network}' in your truffle.js. For an example, see http://truffleframework.com/docs/advanced/configuration`
+    )
+  }
+
+  const { provider, host, port } = truffleNetwork
+  return {
+    ...truffleNetwork,
+    name: network,
+    provider: provider
+      ? typeof provider === 'function'
+        ? provider()
+        : provider
+      : new Web3.providers.WebsocketProvider(
+          host && port ? `ws://${host}:${port}` : `http://localhost:8545`
```

Comment:
I'm a bit worried that this will not work with http providers

Reply:
When the user exclusively defines the `host` and `port` is there any other key to be used to guess the protocol?

Reply:
Let's make sure to asses this in future tests